tests/communication/CMakeLists.txt

```diff
@@ -52,4 +52,7 @@ target_include_directories(app PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
 # Target
 # ######################################################################################################################
 
-target_sources(app PRIVATE ../../src/communication/communication.cpp src/main.cpp)
```

Comment:
I think it would be better to link library under testing instead of adding source files.

Reply:
Which library do you mean?

Reply:
For current project I would suggest creating "communication" static library & link it to app & to the test. In this case it wouldn't be necessary to duplicate required source files & compiler options